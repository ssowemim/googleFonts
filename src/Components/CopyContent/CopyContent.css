.CopyContent {
  position: relative; }
  .CopyContent .fa {
    border: 1px #000 solid;
    padding: 3px; }
    .CopyContent .fa:hover {
      background: #000;
      color: #fff; }
      .CopyContent .fa:hover + .CopyContent__TitlePopUp {
        display: block; }
  .CopyContent .CopyContent__TitlePopUp {
    position: absolute;
    width: 85px;
    top: 20px;
    right: 0;
    text-align: center;
    padding: 5px 2px;
    box-sizing: content-box;
    background: #000;
    color: #fff;
    border-radius: 2px;
    display: none; }
