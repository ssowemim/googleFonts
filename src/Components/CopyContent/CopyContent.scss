.CopyContent {
  position: relative;
  .fa {
    border: 1px #000 solid;
    padding: 3px;
    &:hover {
      background: #000;
      color: #fff;
      + .CopyContent__TitlePopUp {
        display: block;
      }
    }
  }

  .CopyContent__TitlePopUp {
    position: absolute;
    width: 85px;
    top: 20px;
    right: 0;
    text-align: center;
    padding: 5px 2px;
    box-sizing: content-box;
    background: #000;
    color: #fff;
    border-radius: 2px;
    display: none;
  }
}