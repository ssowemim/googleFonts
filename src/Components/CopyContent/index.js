import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './CopyContent.css'

class CopyContent extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
        this.handleClickMouseLeave = this.handleClickMouseLeave.bind(this);
        this.state = {
            titleText : "Click to Copy"
        }
    }

    handleClick(event) {
        this.props.onClick(event.target.dataset.item);
        this.setState({
            titleText: "Copied"
        });
    }

    handleClickMouseLeave() {
        this.setState({
            titleText: "Click to Copy"
        });
    }

    render() {
        return (
            <div className="CopyContent">
                <i className="fa fa-copy"
                    onClick={this.handleClick}
                    onMouseLeave={this.handleClickMouseLeave}
                    data-item={this.props.content}>
                </i>
                <span className="CopyContent__TitlePopUp">{this.state.titleText}</span>
            </div>
        );
    }
}

CopyContent.propTypes = {
    content: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default CopyContent;