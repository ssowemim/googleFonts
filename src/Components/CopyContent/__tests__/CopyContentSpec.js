import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CopyContent from './../index';

describe('CopyContent', () => {
    it('renders correctly', () => {
        const copyContentProps = {
            content: "Some Content",
            onClick: jest.fn()
        }
        let tree = renderer.create(
            <CopyContent {...copyContentProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handleClickMouseLeave should reset titleText state', () => {
        const copyContentProps = {
            content: "Some Content",
            onClick: jest.fn()
        };
        let wrapper = shallow(<CopyContent {...copyContentProps} />);
        wrapper.instance().state.titleText = "Copied";
        wrapper.find('.fa-copy').simulate("mouseleave");
        expect(wrapper.instance().state.titleText).toEqual("Click to Copy")
    });

    it('handleClick should set titleText state', () => {
        let mockClickHandler = jest.fn();
        let mockObject = {
            target: {
                dataset: {
                    item :"Pankaj"
                }
            }
        }
        const copyContentProps = {
            content: "Some Content",
            onClick: mockClickHandler
        };
        let wrapper = shallow(<CopyContent {...copyContentProps} />);
        expect(wrapper.instance().state.titleText).toEqual("Click to Copy")
        wrapper.find('.fa-copy').simulate("click", mockObject);
        expect(mockClickHandler).toHaveBeenCalledWith(mockObject.target.dataset.item)
        expect(wrapper.instance().state.titleText).toEqual("Copied")
    });
});
