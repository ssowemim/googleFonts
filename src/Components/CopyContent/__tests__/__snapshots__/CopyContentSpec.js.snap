// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyContent renders correctly 1`] = `
<div
  className="CopyContent"
>
  <i
    className="fa fa-copy"
    data-item="Some Content"
    onClick={[Function]}
    onMouseLeave={[Function]}
  />
  <span
    className="CopyContent__TitlePopUp"
  >
    Click to Copy
  </span>
</div>
`;
